build-system/tasks/check-links.js

```diff
@@ -51,68 +51,51 @@ function getMarkdownFiles() {
 async function checkLinks() {
   maybeUpdatePackages();
   const markdownFiles = getMarkdownFiles();
-  const linkCheckers = markdownFiles.map(function(markdownFile) {
-    return runLinkChecker(markdownFile);
-  });
-  return BBPromise.all(linkCheckers).then(function(allResults) {
-    let deadLinksFound = false;
-    const filesWithDeadLinks = [];
-    allResults.map(function(results, index) {
-      // Skip files that were deleted by the PR.
-      if (!fs.existsSync(markdownFiles[index])) {
-        return;
-      }
+  const allResults = await Promise.all(markdownFiles.map(runLinkChecker));
```

Comment:
I now see what you said about this cleanup not materially change the concurrency. Can you confirm by testing locally as follows? (Logs should be separated by file.)
```
gulp check-links --files "contributing/DEVELOPING.md,contributing/getting-started-e2e.md,contributing/getting-started-quick.md"
```

Reply:
[LGTM](https://gist.github.com/alanorozco/92c889bd51868d44398df9606f625fb6)